ui/src/app/devices/views/DeviceDetails/DeviceNetwork/EditInterface/EditInterface.test.tsx

```diff
@@ -0,0 +1,229 @@
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+
+import EditInterface from "./EditInterface";
+
+import { actions as deviceActions } from "app/store/device";
+import deviceSelectors from "app/store/device/selectors";
+import type { DeviceNetworkInterface } from "app/store/device/types";
+import { DeviceIpAssignment } from "app/store/device/types";
+import type { RootState } from "app/store/root/types";
+import {
+  device as deviceFactory,
+  deviceDetails as deviceDetailsFactory,
+  deviceEventError as deviceEventErrorFactory,
+  deviceInterface as deviceInterfaceFactory,
+  deviceState as deviceStateFactory,
+  deviceStatus as deviceStatusFactory,
+  deviceStatuses as deviceStatusesFactory,
+  rootState as rootStateFactory,
+  subnet as subnetFactory,
+  subnetState as subnetStateFactory,
+} from "testing/factories";
+import { submitFormikForm } from "testing/utils";
+
+const mockStore = configureStore();
+
+describe("EditInterface", () => {
+  let state: RootState;
+  let nic: DeviceNetworkInterface;
+  beforeEach(() => {
+    nic = deviceInterfaceFactory();
+    state = rootStateFactory({
+      device: deviceStateFactory({
+        items: [
+          deviceDetailsFactory({
+            interfaces: [nic],
+            system_id: "abc123",
+          }),
+        ],
+        loaded: true,
+        statuses: deviceStatusesFactory({
+          abc123: deviceStatusFactory(),
+        }),
+      }),
+      subnet: subnetStateFactory({
+        items: [subnetFactory({ id: 1 }), subnetFactory({ id: 2 })],
+        loaded: true,
+      }),
+    });
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("displays a spinner if device is not detailed version", () => {
+    state.device.items[0] = deviceFactory({ system_id: "abc123" });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <EditInterface nicId={nic.id} systemId="abc123" closeForm={jest.fn()} />
+      </Provider>
+    );
+    expect(
+      wrapper.find("[data-testid='loading-device-details']").exists()
+    ).toBe(true);
```

Comment:
I find this test misleading. It implies that an element with a`data-testid='loading-device-details` exists.

In reality, it will never appear anywhere in the DOM - [Spinner doesn't accept the `data-testid` prop](https://github.com/canonical-web-and-design/react-components/blob/main/src/components/Spinner/Spinner.tsx#L21) (https://github.com/canonical-web-and-design/react-components/issues/651). 

I'd suggest to check for Spinner component instance, like it's often done in enzyme-style unit tests, or use react testing library to assert the real DOM (I believe unit tests as such provide less value than BDD-style integration tests, but you already know that 😉 ).

Reply:
Thanks, I've added the reference to the Spinner component, but I think the test id is still worthwhile here as we're not doing a shallow compare so there could be other Spinners that could provide false positives.